### PR TITLE
Add dialog backdrop blur theme variable

### PR DESCRIPTION
**Pull request by noxhirsch**
## Proposed change
I've added a theme variable which allows to blur the background behind dialog popups. This creates a much more modern look and puts the dialog more into focus (less distracting background). Default is set to zero - so it isn't a breaking change.
blur_1
blur_3
blur_2

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
themes:
  blur_theme:
    dialog-bg-blur: 5px
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Not all our dialogs use the new `ha-dialog` yet, so this will not work for every dialog. That will raise a lot of questions, we should either not do this until all dialogs are migrated or make it work for the old (`paper-dialog`) too.

**Comment by noxhirsch** [comment]
Thank you for the hint! I'll add it to the `paper-dialog` tomorrow 👍

File: src/components/ha-dialog.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export class HaDialog extends MwcDialog {
         .mdc-dialog {
           --mdc-dialog-scroll-divider-color: var(--divider-color);
           z-index: var(--dialog-z-index, 7);
+          backdrop-filter: blur(var(--dialog-bg-blur, 0));
```
Comment (bramkragten) [inline src/components/ha-dialog.ts:43]:
```suggestion
          backdrop-filter: var(--dialog-backdrop-filter, unset);
```

**Comment by noxhirsch** [comment]
@bramkragten I tried lots of things in the last few hours to get it working with `paper-dialog`, but sadly I don't have a clue (I've never used Polymer before). If the solution is simple, maybe someone else can help here. If the solution isn't simple or is "hacky", I think we shouldn't do that for a "soon" to be removed element.

My suggestion or rather my wish would be, to add it as it is as undocumented variable, because it's already very useful - the only places where it doesn't work (as far as I can see) are in "edit mode" and in the "backend" (like automation editor). 
The same had already been done with `--header-height` which also does not work everywhere (but is good enough to use for the Lovelace UI)

**Comment by bramkragten** [comment]
I think it would be `iron-overlay-backdrop` you would have to target, which is in the root of the DOM.

**Comment by noxhirsch** [comment]
That right - and also about the only thing I found out 😉 My problem was rather that I didn't know where to put it in the frontend code, but I think (aka. hope) I found a good solution.
image

File: src/resources/ha-style.ts [hunk 2]
```diff
@@ -95,6 +95,10 @@ documentContainer.innerHTML = `<custom-style>
       --rgb-text-primary-color: 255, 255, 255;
       --rgb-card-background-color: 255, 255, 255;
 
+      /* change from opacity to rgba for backdrop-filter */
+      --iron-overlay-backdrop-opacity: 1;
+      --iron-overlay-backdrop-background-color: rgba(0, 0, 0, 0.32);
+
```
Comment (bramkragten) [inline src/resources/ha-style.ts:101]: This is something a theme should do, we should not.

**Review by bramkragten — APPROVED** [review]

File: src/resources/ha-style.ts [hunk 3]
```diff
@@ -113,6 +113,10 @@ documentContainer.innerHTML = `<custom-style>
     paper-dialog-scrollable.can-scroll > .scrollable {
       -webkit-overflow-scrolling: touch !important;
     }
+
+    iron-overlay-backdrop {
```
Comment (bramkragten) [inline src/resources/ha-style.ts:117]: Please add a comment that this is for paper dialog

**Review by bramkragten — APPROVED** [review]

**Pull request merged**